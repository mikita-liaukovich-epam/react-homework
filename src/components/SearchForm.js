import React from "react";
import ReactDOM from "react-dom";
import VODCollection from "../views/VODCollection.View";
import ErrorBoundary from "../components/ErrorBoundary";
import { assetsPath, VODList } from "../models/VODCollection.DataModel";

function clickHandler(e) {
  e.preventDefault();

  const rootElement = document.getElementById("root");

  ReactDOM.render(
    <ErrorBoundary>
      <VODCollection assetsPath={assetsPath} VODList={VODList} />
    </ErrorBoundary>,
    rootElement
  );
}

export default function SearchForm() {
  return (
    <form className="search-form">
      <input
        type="text"
        placeholder="What do you want to watch?"
        className="search-text"
      />
      <button type="submit" className="submit-button" onClick={clickHandler}>
        Search
      </button>
    </form>
  );
};
