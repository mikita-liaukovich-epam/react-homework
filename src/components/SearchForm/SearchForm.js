import React from "react"
import { useDispatch } from 'react-redux'
import { useHistory } from "react-router-dom"
import { searchMovie } from '../../redux/actions/Movie.actions'

import './SearchForm.scss';

export default function SearchForm() {
  const history = useHistory();
  const dispatch = useDispatch();

  const clickHandler = (e) => {
    e.preventDefault()
    const value = document.getElementById('searchField').value;
    history.push(`search?value=${value}`)
    dispatch(searchMovie(value));
  }

  return (
    <form className="search-form">
      <input
        type="text"
        placeholder="What do you want to watch?"
        className="search-text"
        id="searchField"
      />
      <button type="submit" className="submit-button" onClick={clickHandler}>
        Search
      </button>
    </form>
  );
};
