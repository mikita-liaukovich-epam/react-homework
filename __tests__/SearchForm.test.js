import React from 'react'
import SearchForm from '../src/components/SearchForm/SearchForm'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer
    .create(<SearchForm />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
