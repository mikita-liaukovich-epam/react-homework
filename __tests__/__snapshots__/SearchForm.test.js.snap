// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="search-form"
>
  <input
    className="search-text"
    id="searchField"
    placeholder="What do you want to watch?"
    type="text"
  />
  <button
    className="submit-button"
    onClick={[Function]}
    type="submit"
  >
    Search
  </button>
</form>
`;
